File: fintechplatform-ui/src/main/java/com/fintechplatform/android/transactions/ui/TransactionsViewComponent.java
package com.fintechplatform.android.transactions.ui;

import com.fintechplatform.android.alert.AlertHelpersModule;
import com.fintechplatform.android.log.LogModule;
import com.fintechplatform.android.money.MoneyHelperModule;
import com.fintechplatform.android.net.NetModule;
import com.fintechplatform.android.transactions.api.TransactionsAPIModule;
import com.fintechplatform.android.transactions.db.TransactionPersistanceDBModule;
import com.fintechplatform.android.transactions.models.TransactionsManagerModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component (modules = {
        TransactionsPresenterModule.class,
        TransactionsAPIModule.class,
        NetModule.class,
        LogModule.class,
        TransactionPersistanceDBModule.class,
        MoneyHelperModule.class,
        AlertHelpersModule.class,
        TransactionsManagerModule.class,
        TransactionDetailUIModule.class
})
public interface TransactionsViewComponent {
    void inject(TransactionsActivity activity);
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/transactions/di/TransactionsViewComponent.kt
package com.fintechplatform.ui.transactions.di

import com.fintechplatform.api.log.LogModule
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.api.transactions.api.TransactionsAPIModule
import com.fintechplatform.ui.alert.AlertHelpersModule
import com.fintechplatform.ui.money.MoneyHelperModule
import com.fintechplatform.ui.transactions.ui.TransactionsFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
    TransactionsPresenterModule::class,
    TransactionsAPIModule::class,
    NetModule::class,
    LogModule::class,
    TransactionPersistanceDBModule::class,
    MoneyHelperModule::class,
    AlertHelpersModule::class,
    TransactionsManagerModule::class
])

interface TransactionsViewComponent {
    fun inject(fragment: TransactionsFragment)
}
